src/platform/common.h

```diff
@@ -203,7 +203,7 @@ struct hwdevice_t {
   /**
    * implementations must take ownership of 'frame'
    */
-  virtual int set_frame(AVFrame *frame, AVBufferRef *hw_frames_ctx) {
+  virtual int set_frame(AVFrame *frame, AVBufferRef *hw_frames_ctx, int target_format) {
     BOOST_LOG(error) << "Illegal call to hwdevice_t::set_frame(). Did you forget to override it?";
```

Comment:
This error message is triggered when `encoder = software` is set (and causes the software encoders to become unavailable).